config/main.py

```diff
@@ -986,6 +988,129 @@ def load(filename, yes):
         log.log_info("'load' executing...")
         clicommon.run_command(command, display_cmd=True)
 
+@config.command('apply-patch')
+@click.argument('patch-file-path', type=str, required=True)
+@click.option('-f', '--format', type=click.Choice([e.name.lower() for e in ConfigFormat]),
+               default=ConfigFormat.CONFIGDB.name.lower(),
+               help='format of config of the patch is either ConfigDb(ABNF) or SonicYang')
+@click.option('-d', '--dry-run', is_flag=True, default=False, help='test out the command without affecting config state')
+@click.option('-v', '--verbose', is_flag=True, default=False, help='print additional details of what the operation is doing')
+@click.pass_context
+def apply_patch(ctx, patch_file_path, format, dry_run, verbose):
+    """Apply given patch of updates to Config. A patch is a JsonPatch which follows rfc6902.
+       This command can be used do partial updates to the config with minimum disruption to running processes.
+       It allows addition as well as deletion of configs. The patch file represents a diff of ConfigDb(ABNF)
+       format or SonicYang format.
+
+       <patch-file-path>: Path to the patch file on the file-system."""
+    try:
+        with open(patch_file_path, 'r') as fh:
+            text = fh.read()
+            patch_as_json = json.loads(text)
+            patch = jsonpatch.JsonPatch(patch_as_json)
+
+        config_format = ConfigFormat[format.upper()]
+
+        GenericUpdater().apply_patch(patch, config_format, verbose, dry_run)
+
+        click.secho("Patch applied successfully.", fg="cyan", underline=True)
+    except Exception as ex:
+        click.secho("Failed to apply patch", fg="red", underline=True)
```

Comment:
>secho [](http://example.com/codeflow?start=14&length=5)

`secho` to stderr if this is a critical error.

Reply:
updated locally

Reply:
updated